### ISSUE TITLE
How to start the server?

### ISSUE THREAD

**Issue by hyperpallium** [xtermjs/xterm.js#1726]
I'm pretty sure there must be a server involved somewhere, since it's a terminal, but the *Getting started* section at https://xtermjs.org/ doesn't mention one. 

For someone who's never used `node` or `npm` before, it's not obvious.

I tried downloading the src, naviagating to the demo directory, and trying `node start.js`, `node server.js`... and also, `npm start`, but all crashed.

**Comment by jerch** [comment]
@hyperpallium You already found the demo folder, thats the only serverish part the repo contains.

**BUT: This is only for testing and development, it should never be used or simply copied over to production systems**. It needs additional knowledge about pseudoterminals, access rights and transport encryption to get the security implications covered in an appropriate way. _You have been warned._

@Tyriar Do we need such a warning in the README again? I know we had one before, not sure why it got removed. We could also alter the demo to an electron/puppeteer only development path, though not sure about that since it will make cross browser testing very inconvenient.

**Comment by Tyriar** [comment]
We have a warning on the demo HTML page, I think this is enough:

https://github.com/xtermjs/xterm.js/blob/a7b051c52f1cb6bc4c02a1786a9d949bd8d46ad7/demo/index.html#L37

> We could also alter the demo to an electron/puppeteer only development path

node-pty has a barebones Electron demo https://github.com/Microsoft/node-pty/tree/master/examples/electron, maybe we should link to that at the bottom of the demo too?

**Comment by hyperpallium** [comment]
@jerch I am trying this locally, offline, without internet access.

... but how do you start the server (for the Getting Started example) \?
Or, how do you make the demo work?

 Was I doing it right, but there's some other problem?

BTW I got the terminado demo from the terminado repository working easily (though the terminado demo on xtermjs.com also didn't work).

I now see that making the Getting Started demo difficult to get started is a deliberate strategy, to avoid people unknowingly giving public access to their machine.

I think the more warnings about this the better, so yes, add one to the page. It's curious that there isn't one. It can't hurt.

**Comment by Tyriar** [comment]
@hyperpallium `npm start` was the right thing, did you run `npm install` beforehand? What was the crash?

Instructions for running the demo are here https://github.com/xtermjs/xterm.js/wiki/Contributing#running-the-demo

**Comment by hyperpallium** [comment]
@Tyriar  Thanks for that! Can I suggest having it in a `readme.txt` in the `demo/` directory (as terminado did - very helpful), or at least a link to it?
BTW nodejs.org says only `npm` is supported, but that wiki link says `yarn`. (awesome that yarn has a logo by shittywatercolor). There's a README.md in the root dir, different again.

I installed it first, `npm install xterm` (that part *is* on the Getting Started page). I also downloaded xtermjs from github, and `cd`ed to the demo dir (I assume that's necessary?). I think brutally simple and complete ELI5 instructions are ideal.

IIRC the error message I saw was something to do with `err` not being defined (can't reinstall right now - nodejs is 70MB). It could well be because I'm running it on termux, on 5.1 android, and there are other version issues... BTW I didn't see the version of node required by xtermjs - I tried on both `8.12.0` and `10.11.0`. 

I found logs for my many attempts, this might be the right one:

```
10 silly lifecycle xterm@3.7.0~start: Args: [ '-c', 'node demo/start' ]
11 silly lifecycle xterm@3.7.0~start: Returned: code: 1  signal: null
12 info lifecycle xterm@3.7.0~start: Failed to exec start script
13 verbose stack Error: xterm@3.7.0 start: `node demo/start`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:182:13)
13 verbose stack     at ChildProcess.<anonymous> (/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:182:13)
13 verbose stack     at maybeClose (internal/child_process.js:962:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
```

I guess devs familar with the node ecosystem wouldn't have any of my confusions.

**Comment by jerch** [comment]
@Tyriar Linking to the electron demo in node-pty sounds good.

@hyperpallium `npm install xterm` installs the drilled down package from the npm repo, imho this does not contain the dev parts like the demo in source code repo.
To get the demo part as well, run:
```
#> git clone https://github.com/xtermjs/xterm.js.git
#> cd xterm.js
#> npm install
#> npm start
```
and point your browser to `localhost:3000`.
